doc/psyad/user_guide/implementation.rst

```diff
@@ -78,174 +78,198 @@ through the rules for each supported statement type.
 
-If this is not the case the the associated PSyclone transformation
-will raise an exception, which will be reported to the user as an
-error when running the psyad script.
+where :math:`A` and :math:`B_i` are active variables, :math:`x` and
+:math:`y_i` are expressions that do not contain any active variables
+and there is no limit on the size of :math:`N`.
 
-If this assignment is shown in matrix form:
+If this is not the case the associated PSyclone transformation will
+raise an exception, which will be reported to the user as an error
+when running the psyad script.
 
-.. code-block:: none
+For illustration, consider the case where there are 3 active variables
+(equivalent to :math:`N=2`). We can then write this case in the
+following form:
 
-    A   x y z   A
-    B = 0 1 0 * B
-    C   0 0 1   C
+.. math::
 
-then the adjoint of the assignment is obtained by transposing the
-matrix:
+    A = xA + yB + zC
 
-.. code-block:: none
+where :math:`A`, :math:`B` and :math:`C` are active variables and
+:math:`x`, :math:`y` and :math:`z` are expressions that do not contain
+active variables.
 
-    A   x 0 0   A
-    B = y 1 0 * B
-    C   z 0 1   C
+If the above example is shown in matrix form, we have:
 
-which gives the following adjoint assignments
+.. math::
 
-.. code-block:: none
+    \begin{bmatrix} A \\ B \\ C \end{bmatrix} = \begin{bmatrix} x & y & z \\ 0 & 1 & 0 \\ 0 & 0 & 1 \end{bmatrix} * \begin{bmatrix} A \\ B \\ C \end{bmatrix}
 
-    C^ = C^ + z*A^
-    B^ = B^ + y*A^
-    A^ = x*A^
+The adjoint of the assignment is obtained by transposing the matrix:
 
-where the ^ indicates that the variable now represents the adjoint value.
+.. math::
 
-Notice that if `x=0` in the general form then the tangent linear code
-writes to `A`, rather than updating it i.e.:
+    \begin{bmatrix} \hat{A} \\ \hat{B} \\ \hat{C} \end{bmatrix} = \begin{bmatrix} x & 0 & 0 \\ y & 1 & 0 \\ z & 0 & 1 \end{bmatrix} * \begin{bmatrix} \hat{A} \\ \hat{B} \\ \hat{C} \end{bmatrix}
 
-.. code-block:: none
+where :math:`\hat{A}` denotes the adjoint of the original active variable :math:`A`. This gives the
+following adjoint assignments:
 
-    A = 0*A + y*B + y*C + ...
+.. math::
+
+    \hat{C} = \hat{C} + z\hat{A} \\
+    \hat{B} = \hat{B} + y\hat{A} \\
+    \hat{A} = x\hat{A}
+
+Notice that if that expression :math:`x` is \math:`0` then the tangent-linear code
+writes to :math:`\hat{A}`, rather than updating it i.e.:
+
+.. math::
+
+    A = 0A + yB + yC
 
 which is:
 
-.. code-block:: none
+.. math::
 
-    A = y*B + z*C + ...
+    A = yB + zC
 
-and its adjoint will set `A^` to zero:
+and its adjoint will set :math:`\hat{A}` to zero:
 
-.. code-block:: none
+.. math::
 
-    C^ = C^ + zA^
-    B^ = B^ + yA^
-    A^ = 0
+    \hat{C} = \hat{C} + z\hat{A} \\
+    \hat{B} = \hat{B} + y\hat{A} \\
+    \hat{A} = 0
 
-Lastly, notice that if `x=y=z=0` then the original tangent linear code sets `A` to zero:
+Lastly, notice that if :math:`x=y=z=0` then the original
```

Comment:
I was always taught that there's no such word as firstly/lastly. How about "finally"?

Reply:
OK.